src/viam/sdk/rpc/dial.cpp

```diff
@@ -148,20 +149,26 @@ std::shared_ptr<ViamChannel> ViamChannel::dial(const char* uri,
     if (opts.entity()) {
         entity = opts.entity()->c_str();
     }
-    char* socket_path = ::dial(
+    char* proxy_path = ::dial(
         uri, entity, type, payload, opts.allows_insecure_downgrade(), float_timeout.count(), ptr);
-    if (socket_path == NULL) {
+    if (!proxy_path) {
         free_rust_runtime(ptr);
         throw Exception(ErrorCondition::k_connection, "Unable to establish connecting path");
     }
 
-    std::string address("unix://");
-    address += socket_path;
+    const std::string localhost_prefix("127.0.0.1");
+    std::string address;
+    if (std::string(proxy_path).find(localhost_prefix) == std::string::npos) {
+        // proxy path is not a localhost address and is therefore a unix domain socket (UDS)
+        address += "unix:";
```

Comment:
There is another instance of `unix://` in this repo at `RobotClient::at_local_socket`, which should probably also move to the new form, as well as a comment in `robot/client.hpp` and in `src/viam/examples/README.md`. Maybe get them all as part of this review if this one is being changed?